### PR TITLE
Allow record ids to be any string instead of just UUIDs (fixes #655)

### LINKED ISSUE #655: Relax record id validation
Do we really need to ensure that posted record ids match a uuid regex?

We can generate a uuid when a record without id is posted, and leave the usage of uuid in our official clients.
But is there any reason to use a different regex that collection and bucket names?

edit: The usecase is the Web sync extension chrome.storage.sync: since any key is accepted, it takes the md5 of the key to "generate" UUIDs. Instead we could let the client push any key as record id.
 
* Related #140

### PR DESCRIPTION
**Pull request by leplatrem**
@glasserc r?

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,12 @@ This document describes changes between each past release.
 3.1.1 (unreleased)
 ==================
 
+**Protocol**
+
+- Allow record ids to be any string instead of just UUIDs (fixes #655).
```
Comment (Natim) [inline CHANGELOG.rst]: nit: ``IDs `

File: docs/api/index.rst [hunk 2]
```diff
@@ -11,7 +11,12 @@ API
 Changelog
 ---------
 
-1.6 (unreleased)
+1.7 (unreleased)
+''''''''''''''''
+
+- Allow record ids to be any string instead of just UUID.
```
Comment (Natim) [inline docs/api/index.rst]: nit: `IDs`

File: kinto/core/storage/generators.py [hunk 3]
```diff
@@ -51,9 +51,5 @@ class UUID4(Generator):
     duplicate would be about 50% (`source <http://en.wikipedia.org/wiki/\
 Universally_unique_identifier#Random_UUID_probability_of_duplicates>`_).
     """
-    regexp = (r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-'
```
Comment (Natim) [inline kinto/core/storage/generators.py]: I don't really like that change :(

Shouldn't we actually inherit from the kinto-core UUID4 generator in kinto to override the regexp with a more general one?

**Comment by Natim** [comment]
r- about removing the regexp from the Generator

**Comment by leplatrem** [comment]
Updated. I also added some tests about posting with existing `id`. I was sure we had some!

**Comment by Natim** [comment]
Oh yeah :)

**Comment by glasserc** [comment]
LGTM

**Pull request merged**